pkg/blockdevice/v1alpha1/blockdevice.go

```diff
@@ -211,6 +212,34 @@ func (bdl *BlockDeviceList) Filter(predicateKeys ...string) *BlockDeviceList {
 	return filteredBlockDeviceList
 }
 
+// GetUsableBlockDevices returns custom blockdevice list which can be used
+func (bdl *BlockDeviceList) GetUsableBlockDevices(spcName, namespace string) (*BlockDeviceList, error) {
+	// Initialize filtered block device list
+	if bdl == nil {
+		return nil, errors.Errorf("failed to get usable blockdevices: blockdevicelist is nil")
```

Comment:
1.Why there is a chance of this function called on `nil` pointer receivers?

Reply:
Not sure why I added this check. I will think about it.

Reply:
removed the check